### DISCUSSION TITLE
Questionnaire ID: preferred form?

### DISCUSSION THREAD

**Discussion by arthur-shaw** [arthur-shaw/susoapi#30] [Ideas]
Different functions request the questionnaire ID in different forms.

Some require the ID as a single string of the form `[qnr_id]$[version]`. For example:

```
start_export(
  qnr_id,
  export_type,
  interview_status = "All",
  from = "",
  to = "",
  access_token = "",
  refresh_token = "",
  storage_type = "",
  translation_id = "",
  include_meta = TRUE,
  server = Sys.getenv("SUSO_SERVER"),
  workspace = Sys.getenv("SUSO_WORKSPACE"),
  user = Sys.getenv("SUSO_USER"),
  password = Sys.getenv("SUSO_PASSWORD")
)
```

Others (most?) require the ID in two parts: questionaire ID and questionnaire version

```
get_assignments(
  search_by = "",
  qnr_id = "",
  qnr_version = "",
  responsible = "",
  supervisor_id = "",
  show_archive = FALSE,
  order = "",
  server = Sys.getenv("SUSO_SERVER"),
  workspace = Sys.getenv("SUSO_WORKSPACE"),
  user = Sys.getenv("SUSO_USER"),
  password = Sys.getenv("SUSO_PASSWORD")
)
```

Meanwhile, the REST and GraphQL APIs are similarly inconsistent. 

In REST, for example, one finds both forms. 

Questionnaire ID as one string:

![image](https://user-images.githubusercontent.com/47256431/176964309-23fcdbd7-f98c-4bad-90c9-418559577b23.png)

Questionnaire ID as two strings:

![image](https://user-images.githubusercontent.com/47256431/176964208-446c569b-997d-4a22-b9c4-1419c7d1050d.png)

In GraphQL, the norm seems to be questionnaire ID in two parts. 

![image](https://user-images.githubusercontent.com/47256431/176964455-6fa34a04-9a5a-4ac4-acd3-9ac61199fcf0.png)

But I may have missed something

Two questions:

1. Do you think `{susoapi}` should adopt a single form for providing the questionnaire ID as an argument?
2. If so, which form do you prefer and why?

**Comment by ashwinikalantri** [comment]
I prefer `qnr_id` and `qnr_ver` as two separate arguments. It is cleaner, and easier to use.
